src/views/TimerSequence/TimerSequence.tsx

```diff
@@ -0,0 +1,200 @@
+        reorderTimers,
+    } = useTimer();
+
+    const [isShareModalOpen, setIsShareModalOpen] = useState(false);
+    const [shareUrl, setShareUrl] = useState('');
+
+    const { encoded } = useParams(); // Check for an encoded parameter in the URL
+
+    useEffect(() => {
+        // On component mount or if `encoded` changes:
+        if (encoded) {
+            // Decode from URL parameter
+            try {
+                const decodedSequence = decodeSequence(decodeURIComponent(encoded));
+                setTimerSequence(decodedSequence);
+            } catch (e) {
+                console.error('Failed to decode sequence from URL:', e);
+                // Fallback to local storage or default
+                const storedSequence = loadFromLocalStorage('timerSequence');
+                if (storedSequence && storedSequence.length > 0) {
+                    setTimerSequence(storedSequence);
+                } else {
+                    setTimerSequence([
+                        { type: 'countdown', label: 'Warm Up', initialTime: 2000, description: '' },
+                        { type: 'tabata', label: 'Stretch', rounds: 2, workDuration: 2000, breakDuration: 1000, description: '' },
+                        { type: 'stopwatch', label: 'Cool Down', initialTime: 2000, description: '' },
+                        { type: 'xy', rounds: 3, label: 'Jumping jacks', roundMinutes: 1, roundSeconds: 20, description: '' },
+                    ]);
+                }
+            }
+        } else {
+            // No encoded parameter: load from storage or default
+            const storedSequence = loadFromLocalStorage('timerSequence');
+            if (storedSequence && storedSequence.length > 0) {
+                setTimerSequence(storedSequence);
+            } else {
+                setTimerSequence([
+                    { type: 'countdown', label: 'Warm Up', initialTime: 2000, description: '' },
+                    { type: 'tabata', label: 'Stretch', rounds: 2, workDuration: 2000, breakDuration: 1000, description: '' },
+                    { type: 'stopwatch', label: 'Cool Down', initialTime: 2000, description: '' },
+                    { type: 'xy', rounds: 3, label: 'Jumping jacks', roundMinutes: 1, roundSeconds: 20, description: '' },
+                ]);
+            }
+        }
+    }, [setTimerSequence, encoded]);
+
+    const currentTimer = sequence[currentIndex];
+
+    const handleShareClick = () => {
+        // Encode current sequence
+        const encodedData = encodeSequence(sequence);
+
+        // Extract the base portion of the URL before "#/sequence"
+        const fullUrl = window.location.href;
+        const baseWithoutHash = fullUrl.split('#/')[0];
+        // baseWithoutHash now contains everything before "#/sequence"
+        // For example: "https://example.com/somePath/"
+
+        // Construct the share URL
+        const url = `${baseWithoutHash}#/sequence/${encodeURIComponent(encodedData)}`;
```

Comment:
With the encoding, the url is a bit ugly. You could consider base64 encoding the data, and using that in the URL. It is still ugly, but perhaps a little less so!